puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/builders/smart-proxy-plugin.yaml

```diff
@@ -0,0 +1,18 @@
+- builder:
+    name: smart-proxy-plugin
+    builders:
+      - shell: |+
```

Comment:
Any particular reason you inline it here rather than as a separate file?

Reply:
To be honest, i don't really understand why we keep the actual shell in a separate file. Our JJB is a little too spread out in my opinion.

I can see having a separate shell file if multiple builders use it, but I'm not sure we're currently doing that with any shell script files. Similarly I don't understand why we have split out builders when lots of them are only used by one job. It seems like unnecessary overcomplication.

Reply:
I think for many YAML files you're absolutely right. In the case of bash scripts it does have the advantage of syntax highlighting.